ckanext/canada/auth.py

```diff
@@ -27,6 +28,17 @@ def datastore_upsert(up_func, context, data_dict):
     return up_func(context, data_dict)
 
 
+@chained_auth_function
+def user_create(up_func, context, data_dict=None):
```

Comment:
@RabiaSajjad you will have to register this auth method in the `get_auth_functions` method of the `DataGCCAInternal` plugin class in the `plugins.py` script.

And since it will only be registered in the internal plugin class, you would not need the check for:
```
if 'canada_internal' in config.get('ckan.plugins')
```

Reply:
Ahhh, another issue here actually...with upstream code. The User view is not using the toolkit (framework) `check_access` method. So our `user_create` auth method will never get called.

Will make a PR against our ckan 2.9 fork/branch to fix this.

Reply:
@RabiaSajjad okay, figure out the issue. Your `user_create` auth method will also need the decorator:
```
auth_allow_anonymous_access
```

Which can be imported from `ckan.plugins.toolkit`